metaflow/plugins/cards/card_server.py

```diff
@@ -50,15 +45,20 @@ class RunWatcher(Thread):
     def __init__(self, flow_name, connection: Connection):
         super().__init__()
 
-        self._watch_file = os.path.join(
-            os.getcwd(), DATASTORE_LOCAL_DIR, flow_name, "latest_run"
+        local_root = LocalStorage.datastore_root
+        if local_root is None:
+            local_root = LocalStorage.get_datastore_root_from_config(
+                lambda _: None, create_on_absent=False
+            )
+        self._watch_file = (
+            os.path.join(local_root, flow_name, "latest_run") if local_root else None
```

Comment:
LGTM but just one quick question : Under what circumstances is the `self._watch_file` set to `None` ie when does `local_root` become none. 

Few things we can do : 

1. We can log to the user that the `local_root` was not found (logging in the init function). 
2. On every call to the `get_run_id` function we can also check if `self._watch_file == None` . One reason for doing this is because users can start the server before they launch the first run of a flow (in which case the file will not be present and `LocalStorage.get_datastore_root_from_config` might return None) . If it's None then we re-run the `local_root` extraction and set the `self._watch_file`.

Reply:
Thanks, done all the changes..